### ISSUE TITLE
Failed to load UTXOs

### ISSUE THREAD

**Issue by moneymanolis** [cryptoadvance/spectrum#21]
Apparently, the `check_utox `method is not working properly in Spectrum, resulting in such errors:

![grafik](https://user-images.githubusercontent.com/70536101/205505205-f1512bdf-4b24-41ec-8e24-d37b9bc534b2.png)

This results in not being able to view the `Transactions` tab or the `Addresses` tab.

**Comment by k9ert** [comment]
First thing is to get a stacktrace by replacing:
try:
...
except Exception as e:
   raise SpecterError(e)
```
with 
...
except Exception as e:
    logger.exception(e)
   raise SpecterErrot(e)
```

This is not a good pattern. Ideally, one should never catch Exception. Ideally, one would remove that try/except block entirely. We can in the meantime, remove these patterns entirely in page-serving controllers but not yet in API/JSON serving controllers as the error-message is not yet guaranteed to bubble up properly.

**Comment by k9ert** [comment]
Most propably fixed with #2026
